File: java/poker/src/main/java/com/duchen/poker/model/type/ThreeTypeChecker.java
package com.duchen.poker.model.type;

import java.util.Arrays;
import java.util.List;

public class ThreeTypeChecker implements TypeChecker {

    @Override
    public CardGroupType getType() {
        return CardGroupType.THREE;
    }

    @Override
    public List<Integer> getMyPossibleCardCount() {
        return Arrays.asList(3);
    }

    @Override
    public int getMaxCardIfIsMyType(List<Integer> cards) {
        if (cards.get(0).intValue() == cards.get(1) && cards.get(1).intValue() == cards.get(2)) {
            return cards.get(0);
        }
        return -1;
    }
}

File: java/poker/src/main/kotlin/poker/model/type/ThreeTypeChecker.kt
package poker.model.type

import java.util.Arrays

class ThreeTypeChecker : TypeChecker {

    override val type: CardGroupType
        get() = CardGroupType.THREE

    override val myPossibleCardCount: List<Int>
        get() = Arrays.asList(3)

    override fun getMaxCardIfIsMyType(cards: MutableList<Int>): Int {
        return if (cards[0] == cards[1] && cards[1] == cards[2]) {
            cards[0]
        } else -1
    }
}
